KubeArmor/BPF/system_monitor.c

```diff
@@ -408,81 +430,55 @@ static __always_inline void get_outer_key(struct outer_key *pokey,
     }
 }
 
-// == Pid NS Management == //
-
-static __always_inline u32 add_pid_ns()
+static __always_inline u32 get_kubearmor_config(u32 config)
 {
-    struct task_struct *task = (struct task_struct *)bpf_get_current_task();
-    u32 one = 1;
-
-#if defined(MONITOR_HOST)
-
-    u32 pid_ns = get_task_pid_ns_id(task);
-    if (pid_ns != PROC_PID_INIT_INO)
+    u32 *value = bpf_map_lookup_elem(&kubearmor_config, &config);
+    if (!value)
     {
         return 0;
     }
 
-    u32 pid = bpf_get_current_pid_tgid() >> 32;
-    if (bpf_map_lookup_elem(&pid_ns_map, &pid) != 0)
-    {
-        return pid;
-    }
-
-    bpf_map_update_elem(&pid_ns_map, &pid, &one, BPF_ANY);
-    return pid;
+    return *value;
+}
 
-#else // MONITOR_CONTAINER or MONITOR_CONTAINER_AND_HOST
+// == Pid NS Management == //
 
+static __always_inline u32 add_pid_ns()
+{
+    struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+    u32 one = 1;
     u32 pid_ns = get_task_pid_ns_id(task);
-    if (pid_ns == PROC_PID_INIT_INO)
+    if (get_kubearmor_config(_MONITOR_HOST) && pid_ns == PROC_PID_INIT_INO)
```

Comment:
I think we should reverse the checks for performance reasons.
```suggestion
    if (pid_ns == PROC_PID_INIT_INO && get_kubearmor_config(_MONITOR_HOST))
```